src/middlewared/middlewared/plugins/interface/configure.py

```diff
@@ -137,14 +137,18 @@ def configure(self, data, aliases, wait_dhcp=False, options=None):
                     self.logger.debug('{}: removing possible valid_lft and preferred_lft on {}'.format(name, addr))
                     iface.replace_address(addr)
 
-        # carp must be configured after removing addresses
-        # in case removing the address removes the carp
```

Comment:
This comment is still valid, no?

Reply:
I don't understand the comment honestly. This is configuring CARP on newly created interfaces. It's not creating CARP after removing addresses. Do we allow the ability to remove IP addresses without removing the virtual IP?

Reply:
This code does not only run on new interfaces, it used to sync interface which means addresses can be removed and CARP needs to be reconfigured

Reply:
I don't see how that's possible because when you remove an interface that means it already has an `advskew` set. So on line 110, we break out of this method because `advskew` gets set to something.

Reply:
Not removing the interface, removing an alias